src/providers/dataProvider/dataprovider.test.ts

```diff
@@ -0,0 +1,81 @@
+import { type DataProvider } from 'react-admin'
+import { getDataProvider } from '.'
+import { R_BATCHES } from '../../constants'
+import { DateTime } from 'luxon'
+
+const year: number = 2023
+
+const generateDummyBatchForTesting = (name?: string): Omit<Batch, 'id'> => {
+  return {
+    name: name ?? 'Dummy-Batch',
+    startDate: DateTime.now().toFormat('yyyy-MM-dd'),
+    endDate: DateTime.now().plus({ day: 1 }).toFormat('yyyy-MM-dd'),
+    projectCode: '1',
+    batchNumber: 'V01/2023',
+    yearOfReceipt: String(year),
+    department: 1,
+    project: 1,
+    platform: 1,
+    organisation: 1,
+    maximumProtectiveMarking: 1,
+    remarks: 'remarks-1',
+    receiptNotes: 'receipt-notes-1',
+    createdAt: DateTime.now().toFormat('yyyy-MM-dd'),
+    createdBy: 1
+  }
+}
+
+describe('CRUD operations on each resource', () => {
+  let provider: DataProvider
+  beforeAll(async () => {
+    provider = await getDataProvider(false)
+  })
+
+  // id is omitted because it is automatically generated by localforage while inserting
+  describe('CRUD Operation on Batch', () => {
+    let id: number
+    let batch: Batch
+
+    it('should create a batch', async () => {
+      const data = generateDummyBatchForTesting()
+      const result = await provider.create<Batch>(R_BATCHES, {
+        data
+      })
+      id = result.data.id
+      batch = result.data
+      expect(id).toBeDefined()
+      expect(batch).toBeDefined()
+    })
+
+    it('should get the batch', async () => {
+      const data = generateDummyBatchForTesting()
+      const result = await provider.getOne<Batch>(R_BATCHES, { id })
+      const { id: omittedId, ...expectedResult } = result.data
+      expect(expectedResult).toMatchObject(data)
+    })
+
+    it('should update the batch', async () => {
+      const updatedValue = generateDummyBatchForTesting('dummy-batch')
+      const result = await provider.update<Batch>(R_BATCHES, {
+        id,
+        previousData: batch,
+        data: {
+          name: 'dummy-batch',
+          createdAt: DateTime.now().toFormat('yyyy-MM-dd')
+        }
+      })
+      const { id: omittedId, ...expectedResult } = result.data
```

Comment:
why is the id omitted?  
It should match the one passed in, shouldn't it?

Reply:
The id is automatically generated by localforage when a new item is inserted. We can pass the id but I thought localforage should handle it iteself.

Reply:
Again - this is `update`, not `create`.  You have copy/pasted the test for `create` 🤣

Reply:
For create the name is `Dummy-Batch` (capitalized). But when we update it is `dummy-batch` (small letters) and to verify the updated value we generate the batch by passing the name `const updatedValue = generateDummyBatchForTesting('dummy-batch')
` (small letters)

Reply:
@TahaKhanAbdalli - I may be mistaken, but I still don't think we should omit the id.  If the id has changed, it is an indication that `update` (or `get`) is broken.